### PR TITLE
gh-104469 Convert _testcapi/exceptions to use AC

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-104469

### CODE REVIEW

File: Modules/_testcapi/exceptions.c [hunk 1]
```diff
@@ -35,16 +48,19 @@ err_restore(PyObject *self, PyObject *args) {
     return NULL;
 }
 
-/* To test the format of exceptions as printed out. */
+/*[clinic input]
+_testcapi.exception_print
+    exception as exc: object
+    legacy: bool = False
```
Comment (sobolevn) [inline Modules/_testcapi/exceptions.c:54]: I think that `bool` is better, it used to be `"i"` legacy converter, which is just `int`.

Thanks! 👍

File: Modules/_testcapi/exceptions.c [hunk 2]
```diff
@@ -59,72 +75,70 @@ exception_print(PyObject *self, PyObject *args)
...
+                                       PyObject *dict)
+/*[clinic end generated code: output=439f0d963c1ce2c4 input=23a73013f8a8795a]*/
+{
     return PyErr_NewExceptionWithDoc(name, doc, base, dict);
 }
 
+/*[clinic input]
+_testcapi.exc_set_object
+    exception as exc: object
+    obj: object
```
Comment (sobolevn) [inline Modules/_testcapi/exceptions.c:100]:
```suggestion
    obj: object
    /
```

?

**Comment by corona10** [comment]
@sobolevn Thanks! Updated!

File: Modules/_testcapi/exceptions.c [hunk 3]
```diff
@@ -142,21 +157,33 @@ raise_exception(PyObject *self, PyObject *args)
     return NULL;
 }
 
-/* reliably raise a MemoryError */
+/*[clinic input]
+_testcapi.raise_memoryerror
+
+reliably raise a MemoryError
```
Comment (erlend-aasland) [inline Modules/_testcapi/exceptions.c]: IMO, the docstring is unneeded.

```suggestion
```

File: Modules/_testcapi/exceptions.c [hunk 4]
```diff
@@ -142,21 +157,33 @@ raise_exception(PyObject *self, PyObject *args)
...
     return PyErr_NoMemory();
 }
 
+/*[clinic input]
+_testcapi.fatal_error
+    message: str(accept={robuffer})
+    release_gil: bool = False
+    /
+
+fatal_error(message, release_gil=False): call Py_FatalError(message)
```
Comment (erlend-aasland) [inline Modules/_testcapi/exceptions.c]:
```suggestion
```

**Comment by corona10** [comment]
@erlend-aasland Thanks! PTAL one more time!

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**